sdk/resourcemanager/Azure.ResourceManager/src/RehydrationOperation.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Azure.Core;
+
+namespace Azure.ResourceManager
+{
+    internal class RehydrationOperation : ArmOperation
+    {
+        private readonly NextLinkOperationImplementation _nextLinkOperation;
+        private readonly OperationInternal _operation;
+
+        public RehydrationOperation(NextLinkOperationImplementation nextLinkOperation, OperationState operationState, ClientOptions? options = null)
+        {
+            _nextLinkOperation = nextLinkOperation;
+            _operation = operationState.HasCompleted
+                ? new OperationInternal(operationState)
+                : new OperationInternal(nextLinkOperation, new ClientDiagnostics(options ?? ClientOptions.Default), operationState.RawResponse);
+        }
+
+        public override string Id => _nextLinkOperation.OperationId;
+
+        public override RehydrationToken? GetRehydrationToken() => _nextLinkOperation?.GetRehydrationToken();
+
+        public override bool HasCompleted => _operation.HasCompleted;
+
+        public override Response GetRawResponse() => _operation.RawResponse;
+
+        public override Response UpdateStatus(CancellationToken cancellationToken = default) => _operation.UpdateStatus(cancellationToken);
+
+        public override ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) => _operation.UpdateStatusAsync(cancellationToken);
+
+        private static void AssertNotNull<T>(T value, string name)
+        {
+            if (value is null)
+            {
+                throw new ArgumentNullException(name);
+            }
+        }
```

Comment:
I do not see this method gets used, maybe we should remove it

Reply:
removed